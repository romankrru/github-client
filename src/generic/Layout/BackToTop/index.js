// @flow

import React from 'react';
import _ from 'lodash';
import { compose, branch, renderNothing, lifecycle, withStateHandlers, withHandlers } from 'recompact';

import { withTransitionState } from '../../hoc';
import styles from './assets/index.css';
import type { TTransitionState } from '../../typedefs';

let onScroll;

const BackToTop = (props: {
    toTop: Function,
    transitionState: TTransitionState,
}) => (
    <button
        onClick={props.toTop}

        className={[
            styles.BackToTop,
            styles[`BackToTop-${props.transitionState}`],
        ].join(' ')}
    >
        <div className={styles.text}>Back To Top</div>
    </button>
);

export default compose(
    withStateHandlers(
        { isShown: false },
        { setIsShown: () => (value: boolean) => ({ isShown: value }) },
    ),

    lifecycle({
        componentDidMount() {
            const windowHeight = window.innerHeight;

            onScroll = _.throttle(() => {
                const shouldBeVisible = window.scrollY > (windowHeight / 1.3);

                if (shouldBeVisible && !this.props.isShown) {
                    this.props.setIsShown(true);
                }

                if (!shouldBeVisible && this.props.isShown) {
                    this.props.setIsShown(false);
                }
            }, 100);

            document.addEventListener('scroll', onScroll);
        },

        componentWillUnmount() {
            document.removeEventListener('scroll', onScroll);
        },
    }),

    withHandlers({
        toTop: () => () => window.scrollTo({
            behavior: 'smooth',
            left: 0,
            top: 0,
        }),
    }),

    withTransitionState({ inProp: props => props.isShown }),

    branch(
        props => props.transitionState === 'exited',
        renderNothing,
    ),
)(BackToTop);
