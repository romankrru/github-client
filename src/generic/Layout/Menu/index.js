import React from 'react'
import { Menu } from 'semantic-ui-react'
import { NavLink, Link } from 'react-router-dom';


const MainMenu = props => {
    console.log(props)

    return (
        <Menu>
            <Menu.Item as={NavLink} to="/" name="Home" />
            <Menu.Item as={NavLink} to="/discover" name="Discover"/>

            <Menu.Menu position="right">
                <Menu.Item as={Link} to="/logout" name="Logout" />
            </Menu.Menu>
        </Menu>
    );
};

export default MainMenu;