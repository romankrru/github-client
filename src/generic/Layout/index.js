// @flow

import * as React from 'react';
import { Container } from 'semantic-ui-react';

import Menu from './Menu';
import styles from './assets/index.css';

const Layout = (props: {
    children: React.Node,
}) => (
    <Container className={styles.Container}>
        <Menu />
        {props.children}
    </Container>
);

export default Layout;
