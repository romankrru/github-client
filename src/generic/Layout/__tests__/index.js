import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Layout from '../';

describe('Layout', () => {
    it('should render correctly', () => {
        const output = shallow(<Layout />);

        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
