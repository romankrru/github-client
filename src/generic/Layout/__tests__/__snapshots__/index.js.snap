// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<Container
  className="Container"
>
  <MainMenu />
</Container>
`;
